apps/total-typescript/src/team/index.tsx

```diff
@@ -25,40 +25,55 @@ const InviteTeam: React.FC<React.PropsWithChildren<InviteTeamProps>> = ({
   setPersonalPurchase,
 }) => {
   const bulkCouponSchema = z
-    .object({maxUses: z.number(), usedCount: z.number()})
-    .transform(({maxUses, usedCount}) => {
+    .object({id: z.string(), maxUses: z.number(), usedCount: z.number()})
+    .nullable()
+    .transform((data, ctx) => {
+      if (data === null) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message:
+            'The purchase is either not a bulk purchase or was queried without its bulk coupon.',
+        })
+        return z.NEVER
+      }
+
+      const {id, maxUses, usedCount} = data
       return {
+        bulkCouponId: id,
         maxUses,
         usedCount,
         numberOfRedemptionsLeft: maxUses - usedCount,
-        redemptionsLeft: maxUses > usedCount,
+        hasRedemptionsLeft: maxUses > usedCount,
       }
     })
 
-  const bulkCouponData = bulkCouponSchema.parse(purchase.bulkCoupon)
+  const {
+    bulkCouponId,
+    maxUses,
+    usedCount,
+    numberOfRedemptionsLeft,
+    hasRedemptionsLeft,
+  } = bulkCouponSchema.parse(purchase.bulkCoupon)
 
   const [canRedeem, setCanRedeem] = React.useState(
-    Boolean(bulkCouponData.redemptionsLeft && !existingPurchase),
+    Boolean(hasRedemptionsLeft && !existingPurchase),
   )
   const userEmail = session?.user?.email
-  const bulkCouponId = purchase?.bulkCoupon?.id
 
   return (
     <>
       <p className="py-3">
-        You have{' '}
-        <strong>{bulkCouponData.numberOfRedemptionsLeft} seats left</strong>.{' '}
-        {bulkCouponData.redemptionsLeft &&
+        You have <strong>{numberOfRedemptionsLeft} seats left</strong>.{' '}
+        {hasRedemptionsLeft &&
```

Comment:
Extracted the individual values from the data object above the return so that they can be used directly and less verbosely here.